core/string/ustring.h

```diff
@@ -521,6 +519,14 @@ class String {
 	char32_t unicode_at(int p_idx) const;
 
 	CharString ascii(bool p_allow_extended = false) const;
+	CharString latin1() const { return ascii(true); }
+	void parse_latin1(const StrRange<char> &p_cstr);
+	static String latin1(const StrRange<char> &p_string) {
```

Comment:
Shouldn't it return a CharString (like the `latin1` above) ? Or maybe be called `parse_latin1` instead ?
Because that would mean that : 
```
String str("Hello");
CharString s1 = str.latin1();
String s2 = String::latin1("Hello"); // This seems like a shorthand for above, but it is not, one is CharString and the other is String
```

Wouldn't it make more sense to have it return a CharString, and maybe have a static parse_latin1 too ?

Reply:
Well I see that utf8 has the same behaviour, so at least it is consistent.

Reply:
I agree, the naming is unnecessarily confusing. I opted it to be consistent for now but agree renaming the bunch may be a good idea.